firestore/lib/src/restaurant_page.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/material.dart';
+import 'package:friendlyeats_flutter_firestore/src/widgets/restaurant_page_review_list.dart';
+
+import 'data/restaurant_provider.dart';
+import 'model/restaurant.dart';
+import 'model/review.dart';
+import 'widgets/dialogs/review_create.dart';
+import 'widgets/restaurant_page_details.dart';
+
+class RestaurantPage extends StatefulWidget {
+  static const route = '/restaurant';
+
+  final Restaurant restaurant;
+
+  RestaurantPage({Key? key, required Restaurant restaurant})
+      : restaurant = restaurant,
+        super(key: key);
+
+  @override
+  _RestaurantPageState createState() => _RestaurantPageState(restaurant);
+}
+
+class _RestaurantPageState extends State<RestaurantPage> {
+  _RestaurantPageState(Restaurant this._restaurant) {
+    _initAsyncData();
+  }
+
+  void _initAsyncData() async {
+    final reviews = await _firestoreRestaurantProvider
+        .getReviewsForRestaurant(_restaurant.id!);
+
+    setState(() {
+      _isLoading = false;
+      _user = FirebaseAuth.instance.currentUser!;
+      _reviews = reviews;
+    });
+  }
+
+  final Restaurant _restaurant;
+  late final User _user;
+  late List<Review> _reviews;
+  bool _isLoading = true;
+  final FirestoreRestaurantProvider _firestoreRestaurantProvider =
+      FirestoreRestaurantProvider();
+
+  void _onCreateReviewPressed(BuildContext context) async {
+    final newReview = await showDialog<Review>(
+      context: context,
+      builder: (BuildContext context) => ReviewCreateDialog(
+        userId: _user.uid,
+        userName: _user.displayName ?? 'Anonymous User',
+      ),
+    );
+    if (newReview != null) {
+      // Save the review
+      await _firestoreRestaurantProvider.addReview(
+        restaurantId: _restaurant.id!,
+        review: newReview,
+      );
+    }
+
+    _updateReviews();
+  }
+
+  void _onAddRandomReviewsPressed() async {
+    // Await adding a random number of random reviews
+    final numReviews = Random().nextInt(5) + 5;
+    for (var i = 0; i < numReviews; i++) {
+      await _firestoreRestaurantProvider.addReview(
+        restaurantId: _restaurant.id!,
+        review: Review.random(
+          userId: _user.uid,
+          userName: _user.displayName ?? 'Anonymous User',
+        ),
+      );
+    }
+
+    _updateReviews();
+  }
+
+  void _updateReviews() async {
+    final reviews = await _firestoreRestaurantProvider
+        .getReviewsForRestaurant(_restaurant.id!);
+    setState(() {
+      _reviews = reviews;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(_restaurant.name),
+      ),
+      floatingActionButton: _isLoading
+          ? null
+          : FloatingActionButton(
+              child: Icon(Icons.add),
+              tooltip: 'Tap to add a review',
+              onPressed: () => _onCreateReviewPressed(context),
+            ),
+      body: Builder(
```

Comment:
This `Builder` made me anticipate finding a `ScaffoldMessenger.of(context)` call, or some such, in the ensuing widgets; but there doesn't seem to be one. Am I being dense, or is it a relic of a previous implementation?

Reply:
I think I may have just forgot to push some changes to this 🙃